tests/mock_unit/test_oob_telemetry.py

```diff
@@ -48,7 +48,7 @@ def test_unit_oob_connection_telemetry(caplog, local_testing_telemetry_setup):
             assert event["UUID"]
             assert datetime.strptime(
                 event["Created_on"], "%Y-%m-%d %H:%M:%S"
-            ) == pytest.approx(now_time, abs=timedelta(seconds=1))
+            ).timestamp() == pytest.approx(now_time.timestamp(), abs=1)
```

Comment:
this test still doesn't look deterministic. Can we mock the time that we use to get `Created_on` and patch in a constant time?

Reply:
it would be tricky to mock as set `Create_on` is set up `datetime.now()` in the code: https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/mock/_telemetry.py#L64.

I can increase the value to make the test more tolerant.

Reply:
unittest.mock lets you patch a particular function. can we try that? https://docs.python.org/3/library/unittest.mock-examples.html#partial-mocking

Reply:
thanks for sharing, let me try

Reply:
changed to use mock and it's working perfectly, new thing learned